### ISSUE TITLE
Vocabulary "mining" mode that automatically selects the next unknown word

### ISSUE THREAD

**Issue by yue-dongchen** [LuteOrg/lute-v3#434]
[As discussed on Discord](https://discord.com/channels/1074759089051160647/1243161537690996777)

Ad hoc point: I don't know if the LingQ implementation that Xolog refers to ("all of the new words in a list") displays surrounding context on that page. The surround context would be quite important, regardless of whether it's realised as a distinct page.

## Discord chat log

**yue_dongchen**:
Particularly for speeches, I always jump through the unknown words before listening to the audio. My proposal is a mode in which the next word would be selected upon saving a term. Does anyone think this would be helpful?

PS No GitHub issue yet. I will create one if there is a prospect of it being implemented

**Xolog9017**
ah yeah that would actually be really nice. I would use it before every comprehensible input video going forward. Because it will make it easy to look for new words in videos and conveniently get them assigned to 1 straight away
Or even better, jz, if you decide to go down this path, you can take some inspiration from LingQ's page where they show all of the new words in a list. Then you don't have to write the code for making the selected word jump to the next unknown word

**jzohrab**
Thanks both 🙃

**Comment by jzohrab** [comment]
Thanks for the issue -- some misc notes:

* I used to have a "shift + arrow = next unknown word" or something like it but it was confusing, so I got rid of it.  :-)  I never got enough value out of this idea, and as the words are already highlighted in blue, it's just a click to the next one.
* Re showing all the new words in a list -- that might be possible, but without context words start to lose meaning.  Maybe that's ok for some people though, depending on learning style.  For long texts/new users, this list would rapidly get out of control.  Perhaps just show them for the current page ?

Lute currently does have a "toggle highlights" thing, where _all_ of the highlighting is hidden.  I wonder if it would suffice to have some variation of this: "toggle highlights except for unknown words" -- that way the new stuff would be highlighted, but the rest of the page would be white/background color.  The wording is clumsy but maybe the idea is ok.  Or maybe the "toggle highlights" thing could do "cycle highlights" -- start as-is, then toggle all off, then toggle all except for the unknown words off.  Of the ideas given, this feels like the smallest amount of work (though it might still be tough, depending on how I wrote that toggle stuff in the past :-) )

**Comment by yue-dongchen** [comment]
Thanks for the reply. I look to submit a PR leveraging the keys "j" and "l" for respectively backwarding and forwording to adjacent unknown words. They're on the home row for QWERTY-like layouts. Concededly not ideal for I guess more geeky users on alternate layouts.

**Comment by jzohrab** [comment]
Good luck on your exams!

**Comment by jzohrab** [comment]
Following up: Lute will soon have customizable shortcuts per #141 , so this could be added as a new shortcut.

Another note: when I originally had this, I found it hard to use.  My eyes couldn't figure out where I was in the page, so it was a bit weird. :-)

**Comment by jzohrab** [comment]
Hi @yue-dongchen - I found that I wanted something like this in some reading as well, so I'm adding new shortcuts to support moving to the prev/next unknown word.  I have those mapped to "shift-arrowleft" and "shift-arrowright" on my mac, it works well.  I also added a small flash for when it moves to the new word, to help me find it.

I'm adding this to the `develop` branch and will release it soon.  cheers!
